### ISSUE TITLE
[Q] discriminator missing in some polymorphic base classes

### ISSUE THREAD

**Issue by aol-nnov** [gigaSproule/swagger-gradle-plugin#125]
I have two class hierarchies:

```
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "__type")
@JsonSubTypes({
        @JsonSubTypes.Type(value = ThingOne.class, name = "ThingOne"),
        @JsonSubTypes.Type(value = ThingTwo.class, name = "ThingTwo")
})
public class Item {
  // ...
}

public class ThingOne extends Item {
  //..
}

// etc
```
and
```
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "__groupType")
@JsonSubTypes({
        @JsonSubTypes.Type(value = GroupOne.class, name = "GroupOne"),
        @JsonSubTypes.Type(value = GroupTwo.class, name = "GroupTwo")
})
public class Group {
  // ...
}

public class GroupOne extends Group {
  //..
}

// etc
```

In `swagger.json` generated by this plugin discriminator is present in type `Item`, but is missing in type `Group`.

If I generate `swagger.json` with SpringFox, discriminator is present in both classes.

What am I missing? Please help me to overcome it..

**Comment by gigaSproule** [comment]
Just for completeness, can you post the relevant code for both those fields and your build.gradle snippet?

**Comment by aol-nnov** [comment]
Benjamin, here is the gradle snippet:
```
buildscript {
  //..
  dependencies {
    classpath 'gradle.plugin.com.benjaminsproule:swagger-gradle-plugin:1.0.4'
  }
}

    swagger {
//    https://github.com/gigaSproule/swagger-gradle-plugin
        apiSource {
            springmvc = true
            locations = ["net.example.api"]
            schemes = ['https']
            host = 'api.example.net'
            basePath = "/${project.name}"

//            attachSwaggerArtifact = true

            info {
                title = project.name
                version = 'v1'
                description = 'This is a sample.'
                termsOfService = 'http://www.example.com/termsOfService'
                contact {
                    email = 'email@email.com'
                    name = 'Name'
                    url = 'http://www.example.com'
                }
                license {
                    url = 'http://www.apache.org/licenses/LICENSE-2.0.html'
                    name = 'Apache 2.0'
                }
            }
            swaggerDirectory = "${project.buildDir}/swagger-ui"
        }
    }
```

Not sure, what should I post for "relevant code for both those fields", though...

**Comment by gigaSproule** [comment]
Field name, annotations, getters setters. Just so when I take a look later, I can hopefully see the problem quickly/recreate it.

**Comment by aol-nnov** [comment]
okay, here goes the model:
```
@MappedSuperclass
@Data
public class Identifiable {
    @Id
    @GenericGenerator(name = "uuid-gen", strategy = "uuid2")
    @GeneratedValue(generator = "uuid-gen")
    @Type(type = "pg-uuid")
    private UUID id;
}

@Entity
@Data
@EqualsAndHashCode(callSuper = true, exclude = {"itemGroup"})
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "__type")
@JsonSubTypes({
        @JsonSubTypes.Type(value = ThingOne.class, name = "ThingOne"),
        @JsonSubTypes.Type(value = ThingTwo.class, name = "ThingTwo")
})
public class Item extends Identifiable implements Serializable {
    private static final long serialVersionUID = 8718410058894901065L;
    @ManyToOne(fetch = FetchType.EAGER)
    @JsonBackReference
    private ItemGroup itemGroup;

    private String description;
}

@Entity
@Data
@EqualsAndHashCode(callSuper = true, exclude = {"slug"})
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "__groupType")
@JsonSubTypes({
        @JsonSubTypes.Type(value = GroupOne.class, name = "GroupOne"),
        @JsonSubTypes.Type(value = GroupTwo.class, name = "GroupTwo")
})
public class ItemGroup extends Identifiable implements Serializable {
    private static final long serialVersionUID = 1999707854594076557L;

    private String name;
    private String slug;

    @OneToMany(mappedBy = "itemGroup", cascade = CascadeType.ALL)
    @JsonManagedReference
    private Set<Item> variants = new HashSet<>();

    private Vendor vendor;

    private String description;
}
```

relevant parts of `swagger.json`

by SpringFox:
```
"definitions": {
    "Item": {
      "type": "object",
      "discriminator": "__type",
      "properties": {
        "description": {
          "type": "string"
        },
        "id": {
          "type": "string",
          "format": "uuid"
        },
        "itemGroup": {
          "$ref": "#/definitions/ItemGroup"
        }
      },
      "title": "Item"
    },
    "ItemGroup": {
      "type": "object",
      "discriminator": "__groupType",
      "properties": {
        "description": {
          "type": "string"
        },
        "id": {
          "type": "string",
          "format": "uuid"
        },
        "name": {
          "type": "string"
        },
        "slug": {
          "type": "string"
        },
        "variants": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/Item"
          }
        },
        "vendor": {
          "$ref": "#/definitions/Vendor"
        }
      },
      "title": "ItemGroup"
    }
  }
```

by your plugin:
```
"Item" : {
      "type" : "object",
      "discriminator" : "__type",
      "properties" : {
        "id" : {
          "type" : "string",
          "format" : "uuid"
        },
        "itemGroup" : {
          "$ref" : "#/definitions/ItemGroup"
        },
        "description" : {
          "type" : "string"
        },
        "fullName" : {
          "type" : "string"
        }
      }
    },
    "ItemGroup" : {
      "type" : "object",
      "properties" : {
        "id" : {
          "type" : "string",
          "format" : "uuid"
        },
        "name" : {
          "type" : "string"
        },
        "slug" : {
          "type" : "string"
        },
        "variants" : {
          "type" : "array",
          "uniqueItems" : true,
          "items" : {
            "$ref" : "#/definitions/Item"
          }
        },
        "vendor" : {
          "$ref" : "#/definitions/Vendor"
        },
        "description" : {
          "type" : "string"
        }
      }
    }
```

hope that helps..

**Comment by gigaSproule** [comment]
Thanks for the extra info

**Comment by gigaSproule** [comment]
This is fixed in 1.0.6. I've just created a test to prove it as well, so a regression shouldn't happen.
